src/Functions/seriesPeriodDetect.cpp

```diff
@@ -0,0 +1,152 @@
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wshadow"
+#pragma clang diagnostic ignored "-Wextra-semi-stmt"
+#pragma clang diagnostic ignored "-Wzero-as-null-pointer-constant"
+#include <pocketfft_hdronly.h>
+#pragma clang diagnostic pop
+
+#include <Functions/IFunction.h>
+#include <Functions/FunctionFactory.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Columns/ColumnsNumber.h>
+#include <Columns/ColumnArray.h>
+
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+    extern const int ILLEGAL_COLUMN;
+}
+
+class FunctionSeriesPeriodDetect : public IFunction
+{
+public:
+    static constexpr auto name = "seriesPeriodDetect";
+
+    static FunctionPtr create(ContextPtr)
+    {
+        return std::make_shared<FunctionSeriesPeriodDetect>();
+    }
+
+    std::string getName() const override
+    {
+        return name;
+    }
+
+    size_t getNumberOfArguments() const override { return 1; }
+
+    bool isVariadic() const override { return false; }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & /*arguments*/) const override
+    {
+        return std::make_shared<DataTypeFloat64>();
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t) const override
+    {
+        ColumnPtr array_ptr = arguments[0].column;
+        const ColumnArray * array = checkAndGetColumn<ColumnArray>(array_ptr.get());
+        const IColumn & src_data = array->getData();
+        //const ColumnArray::Offsets & offsets = array->getOffsets();
+
+        /* const ColumnPtr & col = arguments[0].column;
+        const ColumnVector<Int64> * src_col = checkAndGetColumn<ColumnVector<Int64>>(col.get());
+        const PaddedPODArray<Int64> & values = src_col->getData(); */
+
+        
+
+        auto res = ColumnFloat64::create(1);
+        auto & res_data = res->getData();
+
+        Float64 period;
```

Comment:
This could be good refactoring and I earlier did consider doing it this way. But then I decided to keep it similar to other implementations of `IFunction`  such as [src/Functions/array/arrayDistinct.cpp](url) and [src/Functions/array/arrayReverse.cpp](url)